CalibCalorimetry/EcalTrivialCondModules/src/EcalLaserCondTools.cc

```diff
@@ -0,0 +1,497 @@
+                                        "in the event list file"
+                                     << eventListFileName_ << " at line " << iline << "!\n";
+      //Look for IOV:
+      bool iovFound = true;
+      if (t <= t3) {  //IOV already inserted for previous event.
+        if (verb_ > 1)
+          cout << "Event in same IOV than previous one.\n";
+        continue;
+      }
+
+      while ((iovFound = r.readTime(t1, t2, t3)) && t3 < t) /*NOP*/
+        ;
+
+      if (iovFound) {
+        if (t1 < prevT1 && t3 < prevT3)
+          throw cms::Exception("User")
+              << "IOVs in the correction ascii file are not ordered in increased time as required!\n";
+        else if (t1 < prevT1 || t3 < prevT3)
+          throw cms::Exception("User") << "Found interleaved IOVs in the correction ascii file!\n";
+        processIov(r, t1, t2, t3);
+      } else {
+        cout << "Warning: event beyond last IOV t3. Event time: " << timeToString(t)
+             << ". Last IOV t3: " << timeToString(t3) << "\n";
+      }
+    }
+  } else
+    while (r.readTime(t1, t2, t3)) {
+      ++iIov;
+      if (iIov <= skipIov_) {
+        std::cout << "Skipping IOV " << iIov << "\n";
+        continue;
+      } else if (processedIovs >= nIovs_) {
+        std::cout << "Requested number of IOVs, " << nIovs_ << ", processed.\n";
+        return;
+      } else {
+        processIov(r, t1, t2, t3);
+        ++processedIovs;
+      }
+    }
+}
+
+void EcalLaserCondTools::processIov(CorrReader& r, int t1, int t2[92], int t3) {
+  static int iIov = 0;
+  ++iIov;
+
+  //   FILE* fdebug = 0;
+  //   if(i==) fdebug = fopen("debug.txt", "w");
+
+  //  if(iIov <= skipIov_) { std::cout << "Skipping IOV " << iIov << "\n"; return; }
+
+  const size_t nLmes = 92;
+  cond::Time_t iovStart = 0;
+
+  if (verb_ > 1) {
+    cout << "t1:" << t1 << "(" << timeToString(t1) << ") \n"
+         << "t3: " << t3 << "(" << timeToString(t3) << ")\nt2-t1: ";
+    for (int i = 0; i < 92; ++i)
+      cout << t2[i] - t1 << "\t";
+    cout << "\n";
+  }
+  if (t1 < fromTime_) {
+    std::cout << "Skipping IOV " << iIov << ", "
+              << ", which is before 'fromTime'," << timeToString(fromTime_) << "(" << fromTime_ << ").\n";
+    return;
+  }
+
+  //   if(toTime_!=-1 && t2 < toTime_) { std::cout << "Skipping IOV " << iIov << ", " << ", which is beyond 'toTime',"
+  //                                               << timeToString(toTime_) << "(" << toTime_ << ").\n"; return; }
+
+  if (t1 == 0) {
+    std::cout << "Skipping IOV with t1 = 0"
+              << "\n";
+    return;
+  }
+
+  EcalLaserAPDPNRatios* corrSet = new EcalLaserAPDPNRatios;
+
+  EcalLaserAPDPNRatios::EcalLaserTimeStamp t;
+  iovStart = uint64_t(t1) << 32;
+  for (size_t i = 0; i < nLmes; ++i) {
+    t.t1 = edm::Timestamp(uint64_t(t1) << 32);
+    t.t2 = edm::Timestamp(uint64_t(t2[i]) << 32);
+    t.t3 = edm::Timestamp(uint64_t(t3) << 32);
+    corrSet->setTime(i, t);
+  }
+
+  const int ncrystals = 75848;
```

Comment:
Constants which could be estimated at compile period should be defined with `const expr`.
Please apply the same fix to other constants.

Reply:
Why ? It won't make any difference. The compiler will produce exactly the same code as there is no difficulty in this piece of code to detect that it can be estimated at compile time.

Reply:
This rule is clearly shown as [point 7.5](http://cms-sw.github.io/cms_coding_rules.html#7----design-and-coding-guidelines-1). I tend to fix them given the cost is low.
However, I don't insist on the patch.

Reply:
Rule applied.